crates/contract/src/update.rs

```diff
@@ -199,3 +214,289 @@ fn bytes_used(update: &Update) -> u128 {
+        let mut proposed_updates = ProposedUpdates::default();
+        let update_0 = Update::Contract([0; 1000].into());
+        let bytes_used = bytes_used(&update_0);
+        let update_id_0 = proposed_updates.propose(update_0.clone());
+        assert_eq!(update_id_0.0, 0);
+        let account_id = gen_account_id();
+        assert_eq!(
+            proposed_updates
+                .vote(&update_id_0, account_id.clone())
+                .unwrap(),
+            &HashSet::from([account_id.clone()])
+        );
+
+        proposed_updates.remove_vote(&account_id);
+
+        let expected = TestUpdateVotes {
+            id: 1,
+            entries: BTreeMap::from([(
+                0,
+                UpdateEntry {
+                    update: update_0.clone(),
+                    votes: HashSet::new(),
+                    bytes_used,
+                },
+            )]),
+        };
+
+        let found: TestUpdateVotes = (&proposed_updates).try_into().unwrap();
+        assert_eq!(found, expected);
+    }
+
+    /// asserts that a vote by [`AccountId`] can be changed
+    #[test]
+    fn test_proposed_updates_change_vote() {
+        let mut proposed_updates = ProposedUpdates::default();
+        let update_0 = Update::Contract([0; 1000].into());
+        let update_id_0 = proposed_updates.propose(update_0.clone());
+        assert_eq!(update_id_0.0, 0);
+        let update_1 = Update::Contract([1; 1000].into());
+        let bytes_used = bytes_used(&update_0);
+        let update_id_1 = proposed_updates.propose(update_1.clone());
+        assert_eq!(update_id_1.0, 1);
+
+        let account_id = gen_account_id();
+        assert_eq!(
+            proposed_updates
+                .vote(&update_id_0, account_id.clone())
+                .unwrap(),
+            &HashSet::from([account_id.clone()])
+        );
+
+        // change vote
+        assert_eq!(
+            proposed_updates
+                .vote(&update_id_1, account_id.clone())
+                .unwrap(),
+            &HashSet::from([account_id.clone()])
+        );
+
+        let expected = TestUpdateVotes {
+            id: 2,
+            entries: BTreeMap::from([
+                (
+                    0,
+                    UpdateEntry {
+                        update: update_0.clone(),
+                        votes: HashSet::new(),
+                        bytes_used,
+                    },
+                ),
+                (
+                    1,
+                    UpdateEntry {
+                        update: update_1.clone(),
+                        votes: HashSet::from([account_id.clone()]),
+                        bytes_used,
+                    },
+                ),
+            ]),
+        };
+
+        let found: TestUpdateVotes = (&proposed_updates).try_into().unwrap();
+        assert_eq!(found, expected);
+    }
+
+    /// asserts that a vote for a non-existing update id by [`AccountId`] removes any previous vote by [`AccountId`]
```

Comment:
Should we document this behavior here?

https://github.com/near/mpc/blob/be7acd28c36a3c3d44ebad69c44d72614b578c5e/crates/contract/README.md?plain=1#L264